### PR TITLE
[VarDumper] Dumping exceptions is now more compact

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | no |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

Before:
before

### CODE REVIEW

**Comment by javiereguiluz** [comment]
👍 

This looks much better. Thanks Nicolas!

---

I'd also suggest to not use 1-letter variable names because it makes code hard to understand and maintain in the long term.

**Comment by nicolas-grekas** [comment]
New "after":
after

File: src/Symfony/Component/VarDumper/Caster/ExceptionCaster.php [hunk 1]
```diff
@@ -98,11 +99,13 @@ public static function castTraceStub(TraceStub $trace, array $a, Stub $stub, $is
             return array();
         }
         $lastCall = isset($frames[$i]['function']) ? ' ==> '.(isset($frames[$i]['class']) ? $frames[0]['class'].$frames[$i]['type'] : '').$frames[$i]['function'].'()' : '';
+        $frames[] = array('function' => '');
 
         for ($j += $trace->numberingOffset - $i++; isset($frames[$i]); ++$i, --$j) {
             $call = isset($frames[$i]['function']) ? (isset($frames[$i]['class']) ? $frames[$i]['class'].$frames[$i]['type'] : '').$frames[$i]['function'].'()' : '???';
 
-            $a[Caster::PREFIX_VIRTUAL.$j.'. '.$call.$lastCall] = new FrameStub(
+            $k = $call.$lastCall;
```
Comment (stof) [inline src/Symfony/Component/VarDumper/Caster/ExceptionCaster.php]: please rename these variables to more meaningful names

File: src/Symfony/Component/VarDumper/Caster/ExceptionCaster.php [hunk 2]
```diff
@@ -145,8 +147,13 @@ public static function castFrameStub(FrameStub $frame, array $a, Stub $stub, $is
                 $f['file'] = substr($f['file'], 0, -strlen($match[0]));
                 $f['line'] = (int) $match[1];
             }
+            $relFile = $f['file'];
+            $home = getenv('HOME');
+            if (false !== $home && 0 === strpos($relFile, $home.DIRECTORY_SEPARATOR)) {
+                $relFile = '~'.DIRECTORY_SEPARATOR.substr($relFile, strlen($home) + 1);
```
Comment (stof) [inline src/Symfony/Component/VarDumper/Caster/ExceptionCaster.php]: can be simply ``'~'.substr($relFile, strlen($home))``. No need to remove the directory separator to add it again

File: src/Symfony/Component/VarDumper/Caster/ExceptionCaster.php (hunk 1 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Caster/ExceptionCaster.php]: updated

File: src/Symfony/Component/VarDumper/Caster/ExceptionCaster.php (hunk 2 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Caster/ExceptionCaster.php]: updated

File: src/Symfony/Component/VarDumper/Caster/ExceptionCaster.php [hunk 3]
```diff
@@ -98,33 +99,36 @@ public static function castTraceStub(TraceStub $trace, array $a, Stub $stub, $is
...
                 $trace->keepArgs,
                 true
             );
+            if (array($prefix.'src') === array_keys($f = self::castFrameStub($frame, array(), $frame, true))) {
+                $frame = $f[$prefix.'src'];
+                if (1 === count($frame->value)) {
+                    foreach ($frame->value as $label => $frame) {
+                    }
+                } else {
+                    $label = '';
```
Comment (hhamon) [inline src/Symfony/Component/VarDumper/Caster/ExceptionCaster.php]: Can this statement be moved before the `if` condition to prevent the `else` statement?

Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Caster/ExceptionCaster.php]: updated, no need for any `if/else` in fact

**Comment by nicolas-grekas** [comment]
ping @symfony/deciders

**Comment by nicolas-grekas** [comment]
Satus: needs work
I'd like to try a few more ideas when dumping arguments.

**Pull request merged**